### DISCUSSION TITLE
getAppInstanceId - default prop config option

### DISCUSSION THREAD

**Discussion by gmczoop** [invertase/react-native-firebase#6049] [General]
Very grateful for all of the work that has gone into this repo, and I have a request that I think could help with the Firebase / Google Tag Manager connection. I'm just wondering if it would be possible to add a config boolean where the instanceId can be included as a default prop for all firebase events if true? Just a boolean config, something like includeInstanceIDasDefaultProp = true|false.

**Comment by gmczoop** [comment]
Further to the above, I've tried to use the firebaseAnalytics().setDefaultEventParameters() to set the instanceId, and it works for some events but not the events that trigger on install like 'first_open' or 'session_start'. The root of my problem is I'd like to set the Firebase instanceId as the cid for my GA 360 analytics within GTM. I'm fairly new to app react, and I've tried a few things, but no matter where I set the default props, I cant get any to be passed with these on install events. Any help would be greatly appreciated.

**Reply by mikehardy** [reply]
That's an interesting / subtle point - I'm not sure the instance id is available at that time (when the first open / session start hits are started). This looks like something to bring up with the native SDK teams @ google. I'd pick your most important platform to start (ios or android) then try an issue in firebase-ios-sdk and/or firebase-android-sdk where you ask those teams. Feel free to tag me and cross-link this discussion, I'm curious for the answer, to see if there is anything we can do here or to see if this is a known thing and there is a better way to accomplish your goal

**Reply by gmczoop** [reply]
Thanks for your speedy reply Mike, do you know of a way to delay the initialisation of the firebase analytics until the props are set or would it simply not work due to the loading order? I've tried testing this by switching analytics_auto_collection_enabled to true a bit later after the app has launched, but it doesn't help with setting these props earlier. 

On a separate note, in GTM in the absence of the instanceID a default CID is generated, set and stored somewhere, but there is seemingly no way for me to access this value using any available methods. Would you have any idea where GTM might store generated values? I’m new to app dev so struggling to find the best way to get this value.

**Comment by gmczoop** [comment]
@mikehardy I've raised an issue here: https://github.com/firebase/firebase-ios-sdk/issues/9295
